subnet/mock_registry.go

```diff
@@ -225,7 +250,16 @@ func (msr *MockSubnetRegistry) deleteSubnet(ctx context.Context, network string,
 }
 
 func (msr *MockSubnetRegistry) watchSubnets(ctx context.Context, network string, since uint64) (Event, uint64, error) {
+	msr.mux.Lock()
+	locked := true
+	defer func() {
```

Comment:
The `defer` is there because I had things like `return msr.index` and I needed that expression to be evaluated under lock. Otherwise I would need to introduce a temporary variable.

But looking at it again I see that I'm returning `msr.index` not under lock in one place. So I'll need to reorg the code anyway.